backend/data_controller.py

```diff
@@ -0,0 +1,46 @@
+from google.protobuf.json_format import MessageToDict
+from grpc.messages import data_pb2
+import config
+from flask_restful import Resource
+from flask import request, Flask
+import csv
+from google.protobuf.timestamp_pb2 import Timestamp
+from datetime import datetime
+
+saved_data = {}
```

Comment:
Flask also have a native cache framework that might work here: https://pythonhosted.org/Flask-Cache/

Reply:
I tried to, but protobuf messages cannot be pickled

Reply:
The question is doe you really need caching here? It's a test server with low load, where the OS filesystem cache will give you most of the benefit. Besides, If it had high load, it would crash with out of memory errors as there is no cache eviction.